### PR TITLE
Added an option for users to modify copies of specific ARC settings files

### PR DESCRIPTION
**Pull request by alongd**
To avoid merge conflicts and to facilitate workflows where ARC is installed as a server module for multiple users, each user can now modify their own copy of `settings.py`, `input.py`, and `submits.py`. These three files were relocated into a `settings` folder.
The definitions in the copies may be left partial, and unassigned values will be set to their respective default in ARC.

All users are encouraged to follow this workflow.

The installation instructions in the documentation were also modified.
No tests were added (in fact, all tests do not use this feature, but rather use ARC's internal files)

Thank you for the reviewers who participated in brainstorming these modifications.

### CODE REVIEW

**Comment by alongd** [comment]
@dranasinghe, the tests pass now, it required some more fixes, thanks for alerting me.
It should be ready to review now.
@oscarwumit , @xiaoruiDong , would you like to take a look as well?
Try testing it by creating an `.arc` folder, see the explanations in the documentation.
And thank you all for brainstorming and for this idea!

File: arc/imports.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# Common imports where the user can optionally put a modified copy of an ARC file un their ~/.arc folder
+local_arc_settings_path = os.path.join(os.getenv("HOME"), '.arc', 'settings.py')
+settings = {key: val for key, val in vars(arc_settings).items() if '__' not in key}
+if os.path.isfile(local_arc_settings_path):
+    sys.path.insert(1, local_arc_settings_path)
+    import settings as local_settings
+    settings.update({key: val for key, val in vars(local_settings).items() if '__' not in key})
+    # Set global_ess_settings to None if using a local settings file (ARC's defaults are dummies)
+    settings['global_ess_settings'] = local_settings['global_ess_settings'] or None
```
Comment (dranasinghe) [inline arc/imports.py:21]: This line is problematic. it throws an error for me. `local_settings` not a dictionary.

File: arc/imports.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import arc.settings.settings as arc_settings
+from arc.settings.inputs import input_files
+from arc.settings.submit import submit_scripts
+
+
+# Common imports where the user can optionally put a modified copy of an ARC file un their ~/.arc folder
+local_arc_settings_path = os.path.join(os.getenv("HOME"), '.arc', 'settings.py')
+settings = {key: val for key, val in vars(arc_settings).items() if '__' not in key}
+if os.path.isfile(local_arc_settings_path):
+    sys.path.insert(1, local_arc_settings_path)
```
[the remark this answers is not in the document]
Comment (dranasinghe) [reply arc/imports.py:17]: I sort of fixed the error
```
local_arc_settings_path = os.path.join(os.getenv("HOME"), '.arc')
settings = {key: val for key, val in vars(arc_settings).items() if '__' not in key}
if os.path.isfile(local_arc_settings_path + '/settings.py'):
    sys.path.insert(1, local_arc_settings_path)
    import settings as local_settings
    settings.update({key: val for key, val in vars(local_settings).items() if '__' not in key})
    # Set global_ess_settings to None if using a local settings file (ARC's defaults are dummies)
    #settings['global_ess_settings'] = local_settings['global_ess_settings'] or None
```

**Comment by alongd** [comment]
@dranasinghe, you're right, there was a bug in that line, it indeed isn't a dictionary.
I fixed it (force pushed) and tested - it seems to be working now. Let me know if there are any additional issues

**Comment by dranasinghe** [comment]
Great! It is working out of the box. I think the branch is ready to merge.

**Comment by alongd** [comment]
Great! are the docs modifications good? If so, go ahead and "approve" the code

**Comment by dranasinghe** [comment]
I am approving

**Pull request merged**